msu/utils/math.nut

```diff
@@ -1,4 +1,16 @@
 ::MSU.Math <- {
+	// Clamps an integer or float _value to be within [_min, _max] range
+	// throw InvalidValue exception when _min is greater than _max
+	function clamp(_value, _min, _max)
+	{
+		if (_min > _max)
+		{
+			::logError("_min must be less than or equal to _max");
+			throw ::MSU.Exception.InvalidValue(_min);
+		}
+		return _value < _min ? _min : (_value > _max ? _max : _value);
+	}
```

Comment:
This is definitely a good feature to add. But this implementation does not preserve the type of `_value`. E.g. `clamp(someFloat, 1, 5)` will now convert the float to an integer because I passed integers as min and max. So:
- Either make it preserve the type i.e. the return value should match the type of `_value`.
- Or make this function **always** return an integer (similar to `::Math.max`) and add a separate function `clampf` for floats which always returns a float (similar to `::Math.maxf`).

Perhaps the first option is better? @TaroEld What do you think?